src/axom/quest/detail/Discretize_detail.hpp

```diff
@@ -220,14 +220,15 @@ int discrSeg(const Point2D &a,
     axom::for_all<ExecSpace>(
       curr_lvl_count,
       AXOM_LAMBDA(axom::IndexType i) {
-        out[next_lvl + i * lvl_factor + 0] =
-          new_inscribed_prism(out[curr_lvl + i], Q, T, S, R, pa, pb);
-        out[next_lvl + i * lvl_factor + 1] =
-          new_inscribed_prism(out[curr_lvl + i], U, R, Q, P, pa, pb);
+        OctType *out_ptr = out.data();
+        out_ptr[next_lvl + i * lvl_factor + 0] =
+          new_inscribed_prism(out_ptr[curr_lvl + i], Q, T, S, R, pa, pb);
+        out_ptr[next_lvl + i * lvl_factor + 1] =
+          new_inscribed_prism(out_ptr[curr_lvl + i], U, R, Q, P, pa, pb);
         if(level == 0)
         {
-          out[next_lvl + i * lvl_factor + 2] =
-            new_inscribed_prism(out[curr_lvl + i], S, P, U, T, pa, pb);
+          out_ptr[next_lvl + i * lvl_factor + 2] =
+            new_inscribed_prism(out_ptr[curr_lvl + i], S, P, U, T, pa, pb);
         }
       });
```

Comment:
I'm surprised it's necessary to grab a pointer to out, via `out.ptr()` before changing values in the array. 
That's certainly now how we want to use `ArrayView`.

Would it help if we add the `mutable` keyword? E.g.
```cpp
AXOM_LAMBDA(axom::IndexType i) mutable {
    out[ <some_index> ] = new_inscribed_prism(out[ <other_index> ], <params> );
    ...
}
```

We do that in a few places in axom, e.g.:
* https://github.com/LLNL/axom/blob/e8092581482f648084d82b7d69f20cfd679c88eb/src/axom/spin/tests/spin_bvh.cpp#L1404
* https://github.com/LLNL/axom/blob/e8092581482f648084d82b7d69f20cfd679c88eb/src/axom/quest/DistributedClosestPoint.hpp#L1262

Reply:
@kennyweiss Thanks! `mutable` works here!